fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/notification/NotificationService.java

```diff
@@ -69,56 +83,58 @@ static void addNotificationToPassedMap(String targetFile, Notification notificat
+     * NOTIFICATION_DIRECTORY_PATH.
      */
-    public synchronized void mergeNotificationsIntoSessionForCrossProjectStorage() {
+    public void recordNotifications() {
+        int manualActionCount = 0;
+        File projectParentFile = new File(NOTIFICATION_DIRECTORY_PATH);
         Map<String, Map<String, Notification>> collectorNotifications = NotificationCollector.getNotifications();
-        Map<String, Map<String, Notification>> sessionNotifications = getNotificationsFromSession();
-
-        synchronized (this) {
-            if (sessionNotifications == null) {
-                sessionNotifications = collectorNotifications;
-                Properties userProperties = session.getUserProperties();
-                userProperties.put(NOTIFICATIONS, sessionNotifications);
-            } else {
-                for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
-                    String fileName = entry.getKey();
-                    Map<String, Notification> notificationsForFile = entry.getValue();
-                    for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
-                        addNotificationToPassedMap(fileName, subMapEntry.getValue(), sessionNotifications);
-                    }
+        for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
+            String fileName = entry.getKey();
+            Map<String, Notification> notificationsForFile = entry.getValue();
+            int i = 0;
+            for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
+                File outputFile = new File(projectParentFile, FilenameUtils.getName(fileName + "-" + i++ + ".txt"));
+                try {
+                    FileUtils.forceMkdir(projectParentFile);
+                    FileUtils.write(outputFile, subMapEntry.getValue().getNotificationAsString(), Charset.defaultCharset());
+                    manualActionCount++;
+                } catch (IOException e) {
+                    throw new GenerationException("Could not write manual action notification to disk!", e);
                 }
             }
+
+            if (manualActionCount > 0) {
+                logger.warn("{} manual action notification encountered - details written to {}", manualActionCount,
+                    projectParentFile.getAbsoluteFile());
+            }
+
         }
-    }
 
-    private Map<String, Map<String, Notification>> getNotificationsFromSession() {
-        Properties userProperties = session.getUserProperties();
-        @SuppressWarnings("unchecked")
-        Map<String, Map<String, Notification>> stringMapMap = (Map<String, Map<String, Notification>>) userProperties.get(NOTIFICATIONS);
-        return stringMapMap;
+        NotificationCollector.cleanup();
     }
 
-    void displayNotifications() {
+    void displayNotifications() throws IOException {
```

Comment:
Q: I see it was this way before, but why package-local?

Reply:
Good catch - should be private.